app/controllers/spree/home_controller_decorator.rb

```diff
@@ -0,0 +1,26 @@
+Spree::Admin::HomeController.class_eval do
+  before_filter :set_top_quotes, :set_new_quote, only: :index
+
+  private
+
+
+  def set_top_quotes
+
+    # get quotes with rank
+    top_quotes = Spree::Quote.with_rank.order(rank: :desc).to_a
+
+    # get random quotes
+    random_quotes_count = Spree::Quote.top_quotes_count - top_quotes.count
+    random_quotes = Spree::Quote.published_and_without_rank.sample(random_quotes_count)
+
+    #arrange quotes
+    @top_quotes = Spree::Quote.rank_range.to_a.map do |rank|
```

Comment:
How about this? This will reduce to one loop
```ruby
@top_quotes = Spree::Quote.rank_range.to_a.reduce([]) do |quotes, rank|
  quote = top_quotes.last.try(:rank) == rank ? top_quotes.pop : random_quotes.pop
  quotes << quote if quote
end
```